### PR TITLE
Add active downloads admin filter

### PR DESCRIPTION
**Pull request by MichalW-CP**
Resolves https://github.com/golemfactory/concent/issues/329

### CODE REVIEW

File: concent_api/core/admin.py [hunk 1]
```diff
@@ -1,44 +1,55 @@
...
         return (
-            ('active',  'Active'),
+            ('active', 'Active'),
             ('passive', 'Passive'),
+            ('active_or_downloads', 'Active or with active downloads')
         )
 
     def queryset(self, request, queryset):
+        active_state_names = [x.name for x in Subtask.ACTIVE_STATES]
+        passive_state_names = [x.name for x in Subtask.PASSIVE_STATES]
```
Comment (rwrzesien) [inline concent_api/core/admin.py]: Those two can be placed in constants.

File: concent_api/core/admin.py [hunk 2]
```diff
@@ -1,44 +1,56 @@
...
+from .models import Subtask
 
+ACTIVE_STATE_NAMES = [x.name for x in Subtask.ACTIVE_STATES]
+PASSIVE_STATE_NAMES = [x.name for x in Subtask.PASSIVE_STATES]
 
-class ActivePassiveStateFilter(admin.SimpleListFilter):
 
-    title          = 'Active/passive state'
+class ActivePassiveDownloadsStateFilter(admin.SimpleListFilter):
+    title = 'Active/passive/downloads state'
```
Comment (cameel) [inline concent_api/core/admin.py]: There are a lot of style changes that only distract from real changes. Please put them all in a separate commit.

File: concent_api/core/admin.py [hunk 3]
```diff
@@ -1,44 +1,56 @@
...
         if self.value() == 'active':
-            active_state_names = [x.name for x in Subtask.ACTIVE_STATES]
-            return queryset.filter(state__in=active_state_names)
+            return queryset.filter(state__in=ACTIVE_STATE_NAMES)
         elif self.value() == 'passive':
-            passive_state_names = [x.name for x in Subtask.PASSIVE_STATES]
-            return queryset.filter(state__in=passive_state_names)
+            return queryset.filter(state__in=PASSIVE_STATE_NAMES)
+        elif self.value() == 'active_or_downloads':
+            current_timestamp = get_current_utc_timestamp()
```
Comment (cameel) [inline concent_api/core/admin.py]: What if the value is different than those three?

You should have an `else` and put `assert False` in it so that the code fails if someone adds a new filter and forgets to include it here.

File: concent_api/core/admin.py [hunk 4]
```diff
@@ -48,17 +60,27 @@ class SubtaskAdmin(ModelAdminReadOnlyMixin, admin.ModelAdmin):
         'task_id',
     ]
 
-    def get_provider_public_key(self, obj):  # pylint: disable=no-self-use
+    @classmethod
+    def get_provider_public_key(cls, obj):
```
Comment (cameel) [inline concent_api/core/admin.py]: This change is not related to filters. It's fixing a pylint warning. It should be in a separate commit.

File: concent_api/core/admin.py (hunk 2 above)
Comment (MichalW-CP) [reply concent_api/core/admin.py]: Hmm I haven't seen any in this commit... I've made git diff with master and only one is this above but this change of styling is connected with change in variable value so I dont't think I need to do it in separate commit.

File: concent_api/core/admin.py (hunk 3 above)
Comment (MichalW-CP) [reply concent_api/core/admin.py]: Using ```else``` pycharm warned me that code is unreachable so I put just ```assert False```

File: concent_api/core/admin.py (hunk 1 above)
Comment (MichalW-CP) [reply concent_api/core/admin.py]: Done.

File: concent_api/core/admin.py (hunk 2 above)
Comment (cameel) [reply concent_api/core/admin.py]: There are more:
- Indentation change in imports above
- Removing a space before `'Active'`.

But OK. That's not that much. Let's leave it as is.

File: concent_api/core/admin.py (hunk 3 above)
Comment (cameel) [reply concent_api/core/admin.py]: You could also do:
``` python
assert self.value() in {'active', 'passive', 'active_or_downloads'}
```
at the beginning of the function. And that would actually be more readable, though more repetitive too.

**Comment by cameel** [comment]
Your commit that adds the filter still has some stuff related to pylint warnings:
``` diff
-    def get_subtask_subtask_id(self, obj):  # pylint: disable=no-self-use
+    @classmethod
+    def get_subtask_subtask_id(cls, obj):
```

Also, the description of the commit that fixes warnings should say what you're fixing and why. Right now it's too vague.

File: concent_api/core/admin.py (hunk 4 above)
Comment (cameel) [reply concent_api/core/admin.py]: The commit description is still misleading. I see now that you're not fixing warnings. You're just removing the unnecessary `disable=no-self-use`. So why not describe it as
```
Replace 'pylint: disable=no-self-use' with the use of @classmethod in SubtaskAdmin
```
?
